apps/api/src/auth/auth.service.ts

```diff
@@ -74,6 +75,46 @@ export class AuthService {
     return this.jwtService.sign(token)
   }
 
+  async validateGoogleIdToken(idToken: string) {
+    const client = this.generateGoogleOauthClient()
+    try {
+      // verifyIdToken will throw an error if the token is invalid
+      const ticket = await client.verifyIdToken({
+        idToken,
+        audience: this.configService.get('googleOAuthId'),
+      })
+      const payload = ticket.getPayload() as TokenPayload
+      if (payload.hd !== 'student.chula.ac.th')
+        throw new ForbiddenException('User hosted_domain is not student.chula.ac.th')
+      return payload
+    } catch {
+      return null
```

Comment:
should validate if the thrown error is indeed the error from client.verifyIdToken(). If it isn't, it should be thrown upwards the function call as normal.